src/lib/openapi/spec/archived-feature-schema.ts

```diff
@@ -0,0 +1,103 @@
+import type { FromSchema } from 'json-schema-to-ts';
+
+export const archivedFeatureSchema = {
+    $id: '#/components/schemas/archivedFeatureSchema',
+    type: 'object',
+    additionalProperties: true,
+    required: ['name', 'project'],
+    description: 'An archived project feature flag definition',
+    properties: {
+        name: {
+            type: 'string',
+            example: 'disable-comments',
+            description: 'Unique feature name',
+        },
+        type: {
+            type: 'string',
+            example: 'kill-switch',
+            description:
+                'Type of the flag e.g. experiment, kill-switch, release, operational, permission',
+        },
+        description: {
+            type: 'string',
+            nullable: true,
+            example:
+                'Controls disabling of the comments section in case of an incident',
+            description: 'Detailed description of the feature',
+        },
+        project: {
+            type: 'string',
+            example: 'dx-squad',
+            description: 'Name of the project the feature belongs to',
+        },
+        stale: {
+            type: 'boolean',
+            example: false,
+            description:
+                '`true` if the feature is stale based on the age and feature type, otherwise `false`.',
+        },
+        impressionData: {
+            type: 'boolean',
+            example: false,
+            description:
+                '`true` if the impression data collection is enabled for the feature, otherwise `false`.',
+        },
+        createdAt: {
+            type: 'string',
+            format: 'date-time',
+            nullable: true,
+            example: '2023-01-28T15:21:39.975Z',
+            description: 'The date the feature was created',
+        },
+        archivedAt: {
+            type: 'string',
+            format: 'date-time',
+            nullable: true,
+            example: '2023-01-29T15:21:39.975Z',
+            description: 'The date the feature was archived',
+        },
+        lastSeenAt: {
+            type: 'string',
+            format: 'date-time',
+            nullable: true,
+            deprecated: true,
+            example: '2023-01-28T16:21:39.975Z',
+            description:
+                'The date when metrics where last collected for the feature. This field was deprecated in v5, use the one in featureEnvironmentSchema',
+        },
+        environments: {
+            type: 'array',
+            description:
+                'The list of environments where the feature can be used',
```

Comment:
This is kinda confusing to me. Why is this here and what does it mean? A flag can be used in all environments, right? Is this the environments active in the flag's current project? Do we need this? Can we deprecate it?

Reply:
I originally removed it because in the project archived flags we return empty array but then I found that we also have a deprecated endpoint for listing all archived flags and we put 3 date fields there. I kept the name consistent with other environments fields in other schemas. But this field is only here to stay as long as we have a deprecated endpoint.